libraries/bezier-rs/src/subpath/solvers.rs

```diff
@@ -123,6 +124,118 @@ impl<ManipulatorGroupId: crate::Identifier> Subpath<ManipulatorGroupId> {
+	/// - The new manipulator group to be added
+	/// - The `in_handle` for the first manipulator group of `other`
+	pub(crate) fn round_line_join(&self, other: &Subpath<ManipulatorGroupId>, center: DVec2) -> (DVec2, ManipulatorGroup<ManipulatorGroupId>, DVec2) {
+		let left = self.manipulator_groups[self.len() - 1].anchor;
+		let right = other.manipulator_groups[0].anchor;
+
+		let center_to_right = right - center;
+		let center_to_left = left - center;
+
+		let in_segment = self.get_segment(self.len_segments() - 1).unwrap();
+		let in_tangent = in_segment.tangent(TValue::Parametric(1.));
+
+		let mut angle = center_to_right.angle_between(center_to_left) / 2.;
+		let mut arc_point = center + DMat2::from_angle(angle).mul_vec2(center_to_right);
+
+		if (arc_point - left).angle_between(in_tangent).abs() > PI / 2. {
+			angle = angle - PI * (if angle < 0. { -1. } else { 1. });
+			arc_point = center + DMat2::from_angle(angle).mul_vec2(center_to_right);
+		}
+
+		let center_to_arc_point = arc_point - center;
+
+		// Based on https://pomax.github.io/bezierinfo/#circles_cubic
+		let handle_offset_factor = 4. / 3. * (angle / 4.).tan();
+
+		(
+			left - center_to_left.perp() * handle_offset_factor,
+			ManipulatorGroup::new(
+				arc_point,
+				Some(arc_point + center_to_arc_point.perp() * handle_offset_factor),
+				Some(arc_point - center_to_arc_point.perp() * handle_offset_factor),
+			),
+			right + center_to_right.perp() * handle_offset_factor,
+		)
+	}
+
+	/// Returns the necessary information to create a round cap between the end of `self` and the beginning of `other`.
+	/// The returned items correspond to:
+	/// - The `out_handle` for the last manipulator group of `self`
+	/// - The new manipulator group to be added
+	/// - The `in_handle` for the first manipulator group of `other`
+	pub(crate) fn round_cap(&self, other: &Subpath<ManipulatorGroupId>) -> (DVec2, ManipulatorGroup<ManipulatorGroupId>, DVec2) {
+		// Based on https://pomax.github.io/bezierinfo/#circles_cubic
+		const HANDLE_OFFSET_FACTOR: f64 = 0.551784777779014;
+
+		let left = self.manipulator_groups[self.len() - 1].anchor;
+		let right = other.manipulator_groups[0].anchor;
+
+		let center = (right + left) / 2.;
+		let center_to_right = right - center;
+		let center_to_left = left - center;
+
+		let arc_point = center + center_to_right.perp();
+		let center_to_arc_point = arc_point - center;
+
+		(
+			left - center_to_left.perp() * HANDLE_OFFSET_FACTOR,
+			ManipulatorGroup::new(
+				arc_point,
+				Some(arc_point + center_to_arc_point.perp() * HANDLE_OFFSET_FACTOR),
+				Some(arc_point - center_to_arc_point.perp() * HANDLE_OFFSET_FACTOR),
+			),
+			right + center_to_right.perp() * HANDLE_OFFSET_FACTOR,
+		)
+	}
+
+	/// Returns the two manipulator groups that create a sqaure cap between the end of `self` and the beginning of `other`.
+	pub(crate) fn square_cap(&self, other: &Subpath<ManipulatorGroupId>) -> [ManipulatorGroup<ManipulatorGroupId>; 2] {
```

Comment:
Unless I'm mistaken, I believe the SVG spec allows the distance the square cap extends beyond the outline to be configurable. Could we accomplish this by modifying the `square_cap` variant to wrap an `f64`?  Might be a good idea to add if its relatively easy (no need to include it in the UI demos)

Reply:
I don't think that is part of the SVG spec based on this: https://www.w3.org/TR/SVG2/painting.html#LineCaps.
However, I realized that I did not handle the single point case correctly so I fixed and added tests for that!

Reply:
Looks good!